### PR TITLE
Simplify metadata reference searching in Add-Import.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Telling @MattGertz this is continued work to break #18939 into smaller parts.

**Comment by CyrusNajmabadi** [comment]
retest this please

File: src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs [hunk 1]
```diff
@@ -288,15 +288,30 @@ private static bool IsHostOrTestWorkspace(Project project)
...
+                if (p == project)
+                {
+                    continue;
+                }
+
+                foreach (var reference in p.MetadataReferences)
+                {
+                    if (reference is PortableExecutableReference peReference &&
+                        !IsInPackagesDirectory(peReference) &&
+                        seenReferences.Add(peReference))
```
Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: You've lost the `Distinct` and the check against already being in `seenReferences` with this transformation.

If you don't need `Distinct`, do you still need the comparer support at all?

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: > You've lost the Distinct and the check against already being in seenReferences with this transformation.

I didn't lose the check against being in 'seenReferences'.  When we 'add' it only returns 'true' if the value is not in the collectoin.  So we only ever process a reference once.

Comment (Pilchie) [inline src/Features/Core/Portable/AddImport/AbstractAddImportCodeFixProvider.cs]: Oh, right.

**Pull request merged**